chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/auditdata/ApiKeyActivityAuditDataRetentionService.java

```diff
@@ -0,0 +1,82 @@
+package gov.healthit.chpl.scheduler.job.auditdata;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.sql.SQLException;
+
+import javax.persistence.Query;
+import javax.transaction.Transactional;
+
+import org.postgresql.copy.CopyManager;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+@Component
+public class ApiKeyActivityAuditDataRetentionService extends AuditDataRetentionService {
+    private static final String AUDIT_NAME = "openchpl.api_key_activity";
+
+    @Autowired
+    public ApiKeyActivityAuditDataRetentionService(@Value("${auditDataFilePath}") String auditDataFilePath) {
+        super(auditDataFilePath);
+    }
+
+    @Override
+    public Long getAuditDataCount(Integer month, Integer year) {
+        Query query = entityManager.createQuery(
+                "SELECT count(*) "
+                + "FROM ApiKeyActivityEntity a "
+                + "WHERE MONTH(a.creationDate) = :month "
+                + "AND YEAR(a.creationDate) = :year");
+        query.setParameter("month", month);
+        query.setParameter("year", year);
+
+        return (Long) query.getSingleResult();
+    }
+
+    @Override
+    @Transactional
+    public void deleteAuditData(Integer month, Integer year) {
+        Query query = entityManager.createQuery(
+                "DELETE FROM ApiKeyActivityEntity a "
+                + "WHERE MONTH(a.creationDate) = :month "
+                + "AND YEAR(a.creationDate) = :year");
+        query.setParameter("month", month);
+        query.setParameter("year", year);
+
+        query.executeUpdate();
+    }
+
+    @Override
+    public void archiveDataToFile(Integer month, Integer year, String fileName, boolean includeHeaders) throws SQLException {
+        String copyCmd = "COPY "
+                + "(SELECT * "
+                + "FROM openchpl.api_key_activity "
+                + "WHERE EXTRACT(MONTH FROM creation_date) = " + month.toString() + " "
```

Comment:
Is this safe? Or should it be/can it be done more like a prepared statement with the ":param" sort of notation we use elsewhere in DAOs? Or even written as a function that gets added to the database and just called from here with the parameters?